packages/core/src/llm/base.ts

```diff
@@ -10,11 +10,75 @@ import type {
   LLMCompletionParamsStreaming,
   LLMMetadata,
 } from "./types.js";
+
 import { streamConverter } from "./utils.js";
 
 export abstract class BaseLLM implements LLM {
   abstract metadata: LLMMetadata;
 
+  predict(
+    params: LLMCompletionParamsStreaming,
+  ): Promise<AsyncIterable<CompletionResponse>>;
+  predict(params: LLMCompletionParamsNonStreaming): Promise<CompletionResponse>;
+  async predict(
+    params: LLMCompletionParamsStreaming | LLMCompletionParamsNonStreaming,
+  ): Promise<CompletionResponse | AsyncIterable<CompletionResponse>> {
+    const { prompt, parentEvent, stream } = params;
+
+    if (Array.isArray(prompt)) {
+      if (stream) {
+        const stream = await this.chat({
+          messages: prompt,
+          parentEvent,
+          stream: true,
+        });
+        return streamConverter(stream, (chunk) => {
+          return {
+            text: chunk.delta,
+          };
+        });
+      }
+
+      const chatResponse = await this.chat({
+        messages: prompt,
+        parentEvent,
+      });
+      return { text: chatResponse.message.content as string };
+    }
+
+    if (typeof prompt === "string") {
+      if (stream) {
+        const stream = await this.complete({
+          prompt,
+          parentEvent,
+          stream: true,
+        });
+
+        return stream;
+      }
+
+      return this.complete({
+        prompt,
+        parentEvent,
+      });
```

Comment:
i guess this doesn't work in TS?
```suggestion
      retrun await this.complete({
          prompt,
          parentEvent,
          stream,
        });
```